src/benchmarks/micro/MicroBenchmarks.Mono.props

```diff
@@ -0,0 +1,7 @@
+<Project>
+  <!-- Shared with the autogenerated project -->
+  <Import Project="*Common.props" />
+  <PropertyGroup>
+    <SelfContained>true</SelfContained>
```

Comment:
The `SelfContained` parameter is required in  the `src/BenchmarkDotNet/Templates/MonoAOTLLVMCsProj.txt` template. To have clear separation of responsibility between the performance and BenchmarkDotNet repositories, I recommend avoiding any dependencies if possible.

We can add the property to the `MonoAOTLLVMCsProj.txt` template and import the `$([System.IO.Path]::ChangeExtension('$(OriginalCSProjPath)', '.Common.props'))`.

Reply:
> The `SelfContained` parameter is required in the `src/BenchmarkDotNet/Templates/MonoAOTLLVMCsProj.txt` template. To have clear separation of responsibility between the performance and BenchmarkDotNet repositories, I recommend avoiding any dependencies if possible.

Done. I didn't want to assume how the project gets used.

> We can add the property to the `MonoAOTLLVMCsProj.txt` template and import the `$([System.IO.Path]::ChangeExtension('$(OriginalCSProjPath)', '.Common.props'))`.

Is the suggestion to name the props file as `Common.props` instead of `Mono.props`?
The reason I chose Mono.props is because it allows the benchmarks project to have the settings separate per project type. We could have a `Common.props` imported by template projects, but then we also need some property to indicate the type of project being built. And you would need to set this property in the template project itself. But I don't know enough about how these get used from bdn.

This would be better approach, and if you want to implement it:
1. change the name of Mono.props/targets, and the .Wasm.props/targets to Common.props/targets
2. Set some property to indicate the project type in the template project
3. You also might want to add this props/targets as a common pattern to the other template projects too.
4. in dotnet/performance, add the Common.props and use the project type property to differentiate the settings.
5. Make sure to test the whole thing from `dotnet/performance`, and `dotnet/runtime`.

Reply:
Thanks!

> Is the suggestion to name the props file as Common.props instead of Mono.props?

If we set the SelfContained property in the template and remove the `MicroBenchmarks.Mono.props`, I assume that the `<Import Project="*Common.props" />` should be moved from `MicroBenchmarks.Mono.props`.

Everything else looks good to me and I wouldn't change the naming.

Reply:
We still need the versions set in `Common.props`, and imported into the generated project via `Mono.props` because:
```
Latest BDN copies the PackageReferences from
MicroBenchmarks.csproj to the generated project.

- but it does not copy the version property, thus breaking the build.
- Move the version property to a Common.props, so it can be used by
the main MicroBenchmarks.csproj, and the autogenerated project.
```

re:remove-selfcontained, done 👍